sklearn/model_selection/tests/test_search.py

```diff
@@ -1154,3 +1154,33 @@ def test_search_train_scores_set_to_false():
     gs = GridSearchCV(clf, param_grid={'C': [0.1, 0.2]},
                       return_train_score=False)
     gs.fit(X, y)
+
+
+def test_grid_search_custom_cv_iter():
+    # Check if a one time iterable is accepted as a cv parameter.
+    X, y = make_classification(n_samples=100, random_state=0)
+
+    class CustomSplitter():
+        def __init__(self, n_samples=100):
+            self.indices = KFold(n_splits=5).split(np.ones(n_samples))
+
+        def split(self, X=None, y=None, groups=None):
+            for index in self.indices:
+                yield index
+
+        def get_n_splits(self, X=None, y=None, groups=None):
+            return 5
+
+    gs = GridSearchCV(LinearSVC(random_state=0),
+                      param_grid={'C': [0.1, 0.2, 0.3]}, cv=CustomSplitter())
+    gs.fit(X, y)
+
+    gs2 = GridSearchCV(LinearSVC(random_state=0),
+                       param_grid={'C': [0.1, 0.2, 0.3]}, cv=KFold(n_splits=5))
+    gs2.fit(X, y)
+
+    for key in ('mean_score_time', 'std_score_time',
+                'mean_fit_time', 'std_fit_time'):
+        gs.cv_results_.pop(key)
+        gs2.cv_results_.pop(key)
+    np.testing.assert_equal(gs.cv_results_, gs2.cv_results_)
```

Comment:
The numpy's `testing.assert_equal` is to check equivalence of nested dicts. We don't have an equivalent AFAIK...